src/chains-config/cache/QueryFeeDetailsCache.ts

```diff
@@ -0,0 +1,68 @@
+/**
+ * When checking the cache there are 3 possible choices to return.
+ */
+enum QueryFee {
+	/**
+	 * queryFeeDetails is available
+	 */
+	available = 'available',
+	/**
+	 * queryFeeDetails is not available
+	 */
+	notAvailable = 'notAvailable',
+	/**
+	 * We dont know is queryFeeDetails is available
+	 */
+	unknown = 'unknown',
+}
+
+export class QueryFeeDetailsCache {
+	/**
+	 * Highest known runtime that doesn't have queryFeeDetails.
+	 */
+	private _versionWithoutCall;
+	/**
+	 * Lowest known runtime that has queryFeeDetails
+	 */
+	private _versionWithCall;
+
+	constructor(
+		versionWithoutCall: number | null,
+		versionWithCall: number | null
+	) {
+		this._versionWithoutCall = versionWithoutCall;
+		this._versionWithCall = versionWithCall;
+	}
+
+	public isQueryFeeDetailsAvail(specVersion: number) {
+		const { available, notAvailable, unknown } = QueryFee;
+
+		if (this._versionWithCall && specVersion >= this._versionWithCall) {
+			return available;
+		}
+
+		if (this._versionWithoutCall && specVersion <= this._versionWithoutCall) {
+			return notAvailable;
+		}
+
+		return unknown;
+	}
+
+	public setVersionWithoutCall(specVersion: number) {
+		if (this._versionWithoutCall === null) {
+			this._versionWithoutCall = specVersion;
+		}
+		if (specVersion > this._versionWithoutCall) {
+			this._versionWithoutCall = specVersion;
+		}
```

Comment:
```suggestion
		if (this._versionWithoutCall === null || specVersion > this._versionWithoutCall) {
			this._versionWithoutCall = specVersion;
		}
```

Maybe? Or make the second `if` an `else if` just to be super obvious that it's one or the other.

Reply:
Cool changed them to `else if`s, at first my thought was stacked `if`s themselves look so much cleaner to me. I love the readability of them, but you're right, it makes more sense to just do an `else if` here